python/src/opendp/context.py

```diff
@@ -742,39 +785,74 @@ def __dir__(self):
         """
         return super().__dir__() + list(constructors.keys())  # type: ignore[operator]
 
-    def resolve(self, allow_transformations: bool = False):
-        """Resolve the query into a measurement.
-
-        :param allow_transformations: If true, allow the response to be a transformation instead of a measurement.
+    def resolve(
+        self, 
+        allow_transformations: bool = False, 
+        bounds: Optional[tuple[float, float]] = None, 
+        T: Optional[Type[float]] = None
```

Comment:
I know floats are a supertype of int for typing, but for the `Type` itself, should it be a `Union`? and does it need to be typed `Optional`? Maybe `float` as the default value?

Reply:
Yeah, optional is necessary. The types are distinct though.